Reemit.Common.UnitTests/SharedReaderTests.cs

```diff
@@ -82,66 +82,100 @@ public void ReadBytes_Called_EmptyBufferKeepsRelativeOffsetAndOffset ()
         using var binaryReader = new BinaryReader(stream);
         using var sharedReader = new SharedReader(sharedReaderOffset, binaryReader, new object());
         var actualUnmanagedValues = new T[expectedUnmanagedValues.Length];
+        var isRangeMapped = false;
+        var expectedPositions = new int[expectedUnmanagedValues.Length];
+        var actualRangeMaps = new IRangeMapped[expectedUnmanagedValues.Length];
 
         // Act
         for (var i = 0; i < actualUnmanagedValues.Length; i++)
         {
-            actualUnmanagedValues[i] = readUnmanaged(sharedReader);
+            var expectedPosition = sharedReader.Offset;
+            var actualValue = ((Delegate)readUnmanaged).DynamicInvoke(sharedReader)!;
+
+            if (actualValue is IRangeMapped rangeMapped)
+            {
+                isRangeMapped = true;
+                var actualValueType = actualValue.GetType()!;
```

Comment:
`GetType()` will never be null on a non-null object. The null-forgiving operator can be safely removed.

Reply:
VS has been doing some funky things for me with nullable as of late.

![image](https://github.com/v0idzz/Reemit/assets/7999934/cff33547-2aa4-4d11-9734-747af66aa266)

`actualValue` is an `object?`, but the `null` check should be satisfied above, and I'm not accessing `GetType` via null conditional (`?.`) anyway, so I really don't understand what's going on there. Explicitly adding a `null` check for `actualValue` right before doesn't seem to stop the IDE hints (and sometimes inaccurate warnings) from showing up.

Reply:
Fixed.